### PR TITLE
Notebook to code

### PR DESCRIPTION
**Pull request by kbaran1998**

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ includes:
 
 tasks:
   prepare:
+    - source python-version.sh
```
Comment (kponichtera) [inline Taskfile.yml]: This should be realized through `preconditions`: https://taskfile.dev/usage/#using-programmatic-checks-to-cancel-execution-of-an-task-and-its-dependencies

File: learning_service/src/get_data.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+"""
+Copy data into dataset directory.
+"""
+import os
+import shutil
+
+DATASET_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "dataset")
+DATA_PATH = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), "data")
```
Comment (kponichtera) [inline learning_service/src/get_data.py]: We should externalize both of these to the properties, with the use of Dataconf that Calin implemented (and which is already merged to the master, if I am correct).

File: learning_service/notebooks/requirements.notebook.txt [hunk 3]
```diff
@@ -64,8 +64,8 @@ httplib2==0.19.1
 idna==3.3
 importlib-metadata==4.11.4
 importlib-resources==5.7.1
-ipykernel==6.13.0
-ipython==8.3.0
+ipykernel
```
Comment (kponichtera) [inline learning_service/notebooks/requirements.notebook.txt:67]: Would be great to use requirements freeze to settle on the library versions for both learning and inference services.

**Review by kponichtera — CHANGES_REQUESTED** [review]

File: learning_service/src/get_data.py (hunk 2 above)
Comment (kbaran1998) [reply learning_service/src/get_data.py]: I saw something similar in the inference service in [config directory](https://github.com/remla2022/stackoverflow-tagger/tree/master/interface_service/src/configs) and [config.py file](https://github.com/remla2022/stackoverflow-tagger/blob/master/interface_service/src/config.py) on the `master` branch. Is that what you are describing?

File: learning_service/notebooks/requirements.notebook.txt (hunk 3 above)
Comment (kbaran1998) [reply learning_service/notebooks/requirements.notebook.txt:67]: I froze then in the learning service.

File: learning_service/src/get_data.py (hunk 2 above)
Comment (kponichtera) [reply learning_service/src/get_data.py]: Yes, exactly.

Comment (kbaran1998) [reply learning_service/src/get_data.py]: Added Dynaconf!

File: learning_service/src/text_preprocessing.py [hunk 4]
```diff
@@ -25,8 +26,8 @@
 PREPROCESSOR_DATA_FILE_NAME = "preprocessor_data.joblib"
 PREPROCESSOR_LABELS_FILE_NAME = "preprocessor_labels.joblib"
 
-DATA_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), "dataset")
-OUTPUT_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), "output")
+DATA_PATH = os.path.join(ROOT_DIR, settings.DATASET_FOR_TRAINING_DIR)
+OUTPUT_PATH = os.path.join(ROOT_DIR, settings.OUTPUT_DIR)
```
Comment (kponichtera) [inline learning_service/src/text_preprocessing.py]: There is no need for joining with any kind of root dir - the variables with settings should have absolute/relative path already there.

Also, I know it's not beautiful, but using the enum values can give us at least an illusion of strong typing and highlight whenever a removed property is used or its name  is written with a typo.

```python
DATA_PATH = settings[VarNames.DATASET_FOR_TRAINING_DIR.value]
OUTPUT_PATH = settings[VarNames.OUTPUT_DIR.value]
```

**Review by kponichtera — CHANGES_REQUESTED** [review]

**Review by kponichtera — APPROVED** [review]

**Pull request merged**